test/DTNtupleTPGSimAnalyzer.C

```diff
@@ -374,12 +374,12 @@ void DTNtupleTPGSimAnalyzer::book()
     for (unsigned int i = 0; i < tags.size(); i++){
       m_plots["hPrimsph2Segs" + chambTag]->GetXaxis()->SetBinLabel(i+1, ph2tags[i].c_str());
     }
-    m_plots2["h2DHwQualSegNHits" + chambTag] = new TH2F(("h2DHwQualSegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-1 Segment; Phase-2 Primitive Quality",5,3.5,8.5,5,0.5,5.5);
-    m_plots2["h2DEmuQualSegNHits" + chambTag] = new TH2F(("h2DEmuQualSegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-1 Segment; Phase-2 Emulator Quality",5,3.5,8.5,5,0.5,5.5);
-    m_plots2["h2DTMQualSegNHits" + chambTag] = new TH2F(("h2DTMQualSegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-1 Segment; Phase-1 Primitive Quality",5,3.5,8.5,5,0.5,5.5);
-    m_plots2["h2DHwQualph2SegNHits" + chambTag] = new TH2F(("h2DHwQualph2SegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-2 Segment; Phase-2 Primitive Quality",5,3.5,8.5,5,0.5,5.5);
-    m_plots2["h2DEmuQualph2SegNHits" + chambTag] = new TH2F(("h2DEmuQualph2SegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-2 Segment; Phase-2 Emulator Quality",5,3.5,8.5,5,0.5,5.5);
-    m_plots2["h2DTMQualph2SegNHits" + chambTag] = new TH2F(("h2DTMQualph2SegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-2 Segment; Phase-1 Primitive Quality",5,3.5,8.5,5,0.5,5.5);
+    m_plots2["h2DHwQualSegNHits" + chambTag] = new TH2F(("h2DHwQualSegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-1 Segment; Phase-2 Primitive Quality",5,3.5,8.5,5,-0.5,4.5);
+    m_plots2["h2DEmuQualSegNHits" + chambTag] = new TH2F(("h2DEmuQualSegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-1 Segment; Phase-2 Emulator Quality",5,3.5,8.5,5,-0.5,4.5);
+    m_plots2["h2DTMQualSegNHits" + chambTag] = new TH2F(("h2DTMQualSegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-1 Segment; Phase-1 Primitive Quality",5,3.5,8.5,5,1.5,6.5);
+    m_plots2["h2DHwQualph2SegNHits" + chambTag] = new TH2F(("h2DHwQualph2SegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-2 Segment; Phase-2 Primitive Quality",5,3.5,8.5,5,-0.5,4.5);
+    m_plots2["h2DEmuQualph2SegNHits" + chambTag] = new TH2F(("h2DEmuQualph2SegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-2 Segment; Phase-2 Emulator Quality",5,3.5,8.5,5,-0.5,4.5);
+    m_plots2["h2DTMQualph2SegNHits" + chambTag] = new TH2F(("h2DTMQualph2SegNHits_" + chambTag).c_str(),"; No. of associated hits to the Phase-2 Segment; Phase-1 Primitive Quality",5,3.5,8.5,5,1.5,6.5);
```

Comment:
As the function get_tm_quality_group outputs a value from 2 to 5 (https://github.com/jaimeleonh/DTNtuples/blob/3862f061b69c02f6036047e8c155bdf781548219/test/DTNtupleTPGSimAnalyzer.h#L54), shouldn't the ranges from the TM go from 1.5 to 5.5?

Reply:
I think that should be fixed now